### PR TITLE
Disable interfaces which aren't in use

### PR DESCRIPTION
**Pull request by mdbooth**
I can't install without this patch. deploy always fails with TripleO complaining that the network service failed.

Bonus: this patch auto-detects SR-IOV devices. We can use that to improve SR-IOV support.

### CODE REVIEW

File: playbooks/roles/network_info/tasks/main.yaml [hunk 1]
```diff
@@ -34,3 +34,66 @@
...
+  register: sriov_physfndir
+
+- name: Extract device names for SR-IOV VFs
+  set_fact:
+    sriov_vfs: "{{ sriov_vfs + [ item | dirname | dirname | basename ] }}"
+  loop: "{{ sriov_physfndir.files | map(attribute='path') | list }}"
+
+- name: Get a list of physical network device names
+  set_fact:
+    physical_nics: "{{ physical_nics + [(item | dirname | basename)] }}"
```
Comment (EmilienM) [inline playbooks/roles/network_info/tasks/main.yaml:96]: where do we use this fact?

File: playbooks/roles/network_info/tasks/main.yaml [hunk 2]
```diff
@@ -34,3 +34,66 @@
...
+  find:
+    paths: "{{ device_links }}"
+    patterns:
+    - physfn
+    file_type: link
+  register: sriov_physfndir
+
+- name: Extract device names for SR-IOV VFs
+  set_fact:
+    sriov_vfs: "{{ sriov_vfs + [ item | dirname | dirname | basename ] }}"
```
Comment (EmilienM) [inline playbooks/roles/network_info/tasks/main.yaml:91]: where do we use this fact?

Comment (mdbooth) [reply playbooks/roles/network_info/tasks/main.yaml:91]: In the next commit.

File: playbooks/roles/network_info/tasks/main.yaml (hunk 1 above)
Comment (mdbooth) [reply playbooks/roles/network_info/tasks/main.yaml]: Whoops, looks like a duplicate. I removed it.

**Pull request merged**